### DISCUSSION TITLE
GetMaterialApp.router vs GetMaterialApp

### DISCUSSION THREAD

**Discussion by manikantag** [jonataslaw/getx#1835] [Q&A]
Hi,

How is `GetMaterialApp.router` different from `GetMaterialApp`?

When I override `GetMiddleware.redirectDelegate()`, it was called only when I use `GetMaterialApp.router`, but not when I used `GetMaterialApp`. I need to override this instead of `GetMiddleware.redirect()` as I need to wait for the `AuthController` _(which uses Firebase Auth)_.

Also, I can't register `onGenerateRoute` when using `GetMaterialApp.router` (to address `To use the Navigator API with named routes (pushNamed, pushReplacementNamed, or pushNamedAndRemoveUntil), the Navigator must be provided with an onGenerateRoute handler.` issue when using `Get.toNamed()`)

Thanks,
Manikanta G

**Comment by joseguerracajas** [comment]
do you find a solution? I have the same problem and I can´t solve it :(

**Reply by manikantag** [reply]
No, I didn't get why they behave differently. But I'm using `GetMaterialApp.router` so that `GetMiddleware.redirectDelegate()` works.

To go to other pages, use `Get.rootDelegate.toNamed()` instead of `Get.toNamed()`

**Reply by manikantag** [reply]
Actually I know why they behave differently  - the way `GetMaterialApp.router` constructor calling `super()`. But I didn't know the reason behind this

**Reply by joseguerracajas** [reply]
Perfect, I´ll try using `Get.rootDelegate.toNamed()` instead of `Get.toNamed()` .
Using that do you have the same result as using Navigator.pushnamed()??

**Reply by manikantag** [reply]
I suppose, yes. You mat want to use `Get.rootDelegate.offNamed()` to remove previous page from history so that user can't go back. For example, once user login, use `offNamed()` instead of `toNamed()`to send the user to home page/dashboard. Similarly, logoff.

**Reply by joseguerracajas** [reply]
Get.rootDelegate.toNamed('/route') dont work for me. this code dont execute nothing

**Comment by jonataslaw** [comment]
The first uses the api from Navigator 2, the second from Navigator 1

**Reply by manikantag** [reply]
Here, first means `GetMaterialApp.router` and second is `GetMaterialApp`, am I correct?

**Comment by adekmaulana** [comment]
> Here, first means `GetMaterialApp.router` and second is `GetMaterialApp`, am I correct?


Sorry, i had to bump
is there any fix for app getting black screen everytime it loads if using GetMaterialApp.router, does that happened to you?
